Hw 02

ivankrut856 opened the pull request:
All functional is included.

ivankrut856 committed:
First version.
All functional is included.

ivankrut856 committed:
Wrapper added

jzuken commented on trie/src/main/java/fr/ladybug/trie/Serializable.java:
```diff
@@ -0,0 +1,10 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public interface Serializable {
```
а тут где документация? интерфейсы ещё даже более важно документировать хорошо
-0.5

ivankrut856 replied:
За что снимаете баллы? В памятке не было никакой инфы про это (конкретно про интерфейсы всякие)... Забыл

jzuken commented on trie/src/main/java/fr/ladybug/trie/Trie.java:
```diff
@@ -0,0 +1,205 @@
+    private TrieNode headNode;
+
+    public Trie() {
+        headNode = new TrieNode();
```
лучше инициализировать при объявлении сразу

jzuken commented on trie/src/main/java/fr/ladybug/trie/Trie.java:
```diff
@@ -0,0 +1,205 @@
+        /** Field-wise serialization */
+        @Override
+        public void serialize(OutputStream out) throws IOException {
+            DataOutputStream stream = new DataOutputStream(out);
```
тут тоже можно var

jzuken commented on trie/src/main/java/fr/ladybug/trie/Trie.java:
```diff
@@ -0,0 +1,205 @@
+            stream.writeInt(endingsCount);
+            stream.writeInt(passingThrough);
+            stream.writeInt(links.size());
+            for (Map.Entry<Character, TrieNode> entry : links.entrySet()) {
```
и тут, и ещё много где. вот раз ради таких конструкций var и ввели
-0.5

ivankrut856 replied:
Вы же сказали, что только тогда, когда очевидно из правой части... Здесь в правой части ничего не сказано про типы. Нет, я конечно, поменяю, но какие-то нестабильные требования...

jzuken commented on trie/src/test/java/fr/ladybug/trie/TrieTest.java:
```diff
@@ -0,0 +1,156 @@
+        trie.remove("Hello_ultimate");
+        assertEquals(0, trie.getSize());
+    }
+
```
тесты на исключения? тесты на сериализацию?
-1

ivankrut856 replied:
Блин, а за это тоже теперь снимают? В памятке не было написано же...

jzuken commented on trie/src/main/java/fr/ladybug/trie/Trie.java:
```diff
@@ -0,0 +1,205 @@
+        }
+
+        /** Serialization helper */
+        private void deserializeFromDataStream(DataInputStream stream) throws IOException {
```
а если формат данных неправильный?

ivankrut856 committed:
.travis.yml

ivankrut856 committed:
Fixes

jzuken commented on trie/src/main/java/fr/ladybug/trie/Serializable.java:
```diff
@@ -0,0 +1,25 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
```
однострочные доки так лучше и писать одной строкой: `/** ... */`

jzuken commented on trie/src/main/java/fr/ladybug/trie/Trie.java:
```diff
@@ -0,0 +1,187 @@
+    /** Root of inner tree representation*/
+    private TrieNode headNode = new TrieNode();
+
+    public Trie() {
```
а этот тут зачем?

jzuken commented on trie/src/main/java/fr/ladybug/trie/Trie.java:
```diff
@@ -0,0 +1,187 @@
+     * @param element the element which is to be removed
+     * @return true if element was presented before removal, false otherwise
+     */
+    public boolean remove(String element) {
```
а если передадут null, то всё упадёт. и не только тут. аннотаций нет, никаких проверок тоже
-1

jzuken commented on trie/src/test/java/fr/ladybug/trie/TrieTest.java:
```diff
@@ -0,0 +1,211 @@
+
+        var expected = new ByteArrayOutputStream();
+        var dataExpected = new DataOutputStream(expected);
+        assertDoesNotThrow(() -> dataExpected.writeInt(0));
```
а зачем тут эти проверки? это же не ваша функциональность

ivankrut856 committed:
Explicit empty constructors deleted
Annotations added, as well as org.jetbrains to classpath
Useless checks deleted

jzuken commented:
а сделайте, пожалуйста, конфиг тревиса, чтобы у вас не было конфликтов постоянно в мастере. например, чтобы оно в мастере проверяло все проекты сразу

ivankrut856 committed:
Update .travis.yml

ivankrut856 committed:
Update .travis.yml

ivankrut856 commented:
> а сделайте, пожалуйста, конфиг тревиса, чтобы у вас не было конфликтов постоянно в мастере. например, чтобы оно в мастере проверяло все проекты сразу

Специально посоветовался с ребятами, чтобы понять как лучше. Литвинов разрешает своей группе вносить небольшие изменения каждый раз, чтобы замёржить. У меня тогда будет компромисный вариант. Файлы не конфликтуют только если абсолютно одинаковые, поэтому у меня будет общий файл на все ветки, но тогда некоторые таски у CI будут падать (те, которые не про эту ветку). Я считаю, так можно жить

jzuken commented:
а можно сделать в мастере проверку всех веток сразу, и в каждой ветке только добавлять новую строчку в конфиг. тогда и конфликтов не будет

ivankrut856 committed:
Update .travis.yml

ivankrut856 commented:
> а можно сделать в мастере проверку всех веток сразу, и в каждой ветке только добавлять новую строчку в конфиг. тогда и конфликтов не будет

Сейчас ветка в том состоянии, как я понял ваше предложение. Конфликты всё равно присутствуют. Что делать?

jzuken commented:
добавлять не в середину списка, а в конец, как все делают?

ivankrut856 commented:
> добавлять не в середину списка, а в конец, как все делают?
Я не могу иметь в файле Трэвиса в ветке больше, чем одну задачу, потому что тест будет падать всё равно, поэтому найдётся ветка, у которой строчка не в конце списка в файле в мастере, в чём соль?

ivankrut856 commented:
> добавлять не в середину списка, а в конец, как все делают?

Так. Ща я попробую супер иновационный подход. Don't panic

ivankrut856 committed:
Update .travis.yml

ivankrut856 committed:
Create travis_check

ivankrut856 commented:
> добавлять не в середину списка, а в конец, как все делают?

Так. Я официально сделал универсальный конфиг. Теперь всё работат

ivankrut856 committed:
Update .travis.yml
